sbol2/object.py

```diff
@@ -13,6 +13,32 @@
 from . import validation
 
 
+# This is an internal function, not part of the public API
+def _compare_properties(obj1, obj2):
+    obj1_keys = obj1.properties.keys()
+    if sorted(obj1_keys) != sorted(obj2.properties.keys()):
+        return False
+    # Keys are equal, check values by converting to sets
+    for k in obj1_keys:
+        if set(obj1.properties[k]) != set(obj2.properties[k]):
+            return False
+    return True
+
+
+# This is an internal function, not part of the public API
+def _compare_owned_objects(obj1, obj2):
+    obj1_keys = obj1.owned_objects.keys()
+    if sorted(obj1_keys) != sorted(obj2.owned_objects.keys()):
+        return False
+    # Keys are equal, check values by converting to dicts
+    for k in obj1_keys:
+        oo1 = {oo.identity: oo for oo in obj1.owned_objects[k]}
+        oo2 = {oo.identity: oo for oo in obj2.owned_objects[k]}
+        if oo1 != oo2:
```

Comment:
I was puzzled for a while because I couldn't figure out where the recursion was.  I guess the recursion is implicitly hidden in this comparison operator?  In any case, I see in PR #175 it has been replaced with an explicit call to `compare` which is more clear.